### ISSUE TITLE
Pre-Compiled headers issue

### ISSUE THREAD

**Issue by Chancylvania** [MicrosoftDocs/visualstudio-docs#5891]
Theres more that needs done than just adding the references. You need to add the .cpp file from the original class project to the test project, then turn of pre-compiled headers in the settings to get it to work. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c18bdf17-06e1-aaf2-b6ce-c606c0e260da
* Version Independent ID: bd627b8f-54f0-84e0-55cd-285597af022d
* Content: [Write unit tests for C/C++ - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/test/writing-unit-tests-for-c-cpp?view=vs-2019)
* Content Source: [docs/test/writing-unit-tests-for-c-cpp.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/test/writing-unit-tests-for-c-cpp.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-test**
* GitHub Login: @corob-msft
* Microsoft Alias: **corob**

**Comment by corob-msft** [comment]
@Chancylvania 

Thanks for writing in. Your experience differs from mine, but that might be due to different test frameworks, or different project and solution layouts in use. It's possible to combine the test code with the source under test in a single project, as you indirectly suggest. However, it should also be possible to execute the tests on the objects generated by your source project, without including anything other than the header files in your test cases.

For example, I built a solution with a simple DLL project in C++/CX, and added a Unit Test App for Universal Windows. I added the source project reference to the test project, included the source header file in the test case file, wrote a simple test, and ran it. The tests ran as expected.

I did the same with a solution containing a simple native C++ console app project that exposes a couple of functions in source file `example.cpp` using a header `example.h`, and added a Google Test project. Native code doesn't use references, so I added the header to my test case file, wrote a test, and added the `example.obj` object file to the **Linker** > **Input** > **Additional Dependencies** property in the test project. (I used `$(SolutionDir)$(SolutionName)\$(IntDir)example.obj` as the value, since the project under test had the same name as the solution.) I didn't have to change the precompiled header settings. It just worked.

The details depend on how you architect your tests and which framework you use. Without more detail, I can't determine why your proposal would be required in this article. Perhaps we're being too generic? We do expect people to know how to use the test frameworks they choose, and that assumption might be unwarranted.
